File: src/features/education/domain/CompletionStandard.js
import { createEntity } from '../../../shared/domain/createEntity';

const schema = {
  completionStandardId: { label: '이수기준 ID', required: true },
  researcherCategoryId: { label: '연구활동종사자 분류 ID', required: true },
  requiredHours: { label: '필수 이수시간', required: true, type: 'number', min: 1 },
  effectiveFrom: { label: '기준 적용 시작일', required: true },
  effectiveTo: { label: '기준 적용 종료일', required: true },
};

export function createCompletionStandard(input) {
  const entity = createEntity(input, schema);
  if (entity.effectiveFrom > entity.effectiveTo) {
    throw new Error('기준 적용 시작일은 종료일보다 늦을 수 없습니다.');
  }
  return entity;
}

File: backend/src/main/java/com/oose/labsafety/education/domain/CompletionStandard.java
package com.oose.labsafety.education.domain;

import com.oose.labsafety.common.domain.Identifiable;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public record CompletionStandard(
        @NotBlank(message = "이수기준 ID는 필수입니다.") String completionStandardId,
        @NotBlank(message = "연구활동종사자 분류 ID는 필수입니다.") String researcherCategoryId,
        @Min(value = 1, message = "필수 이수시간은 1 이상이어야 합니다.") int requiredHours,
        @NotBlank(message = "기준 적용 시작일은 필수입니다.") String effectiveFrom,
        @NotBlank(message = "기준 적용 종료일은 필수입니다.") String effectiveTo
) implements Identifiable {
    @Override
    public String id() {
        return completionStandardId;
    }
}
